gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -501,6 +502,14 @@ public static Builder builder() {
   BlobInfo create(BlobInfo blobInfo, byte[] content, BlobTargetOption... options);
 
   /**
+   * Create a new blob.
+   *
+   * @return a complete blob information.
+   * @throws StorageException upon failure
+   */
+  BlobInfo createFromStream(BlobInfo blobInfo, InputStream content, BlobTargetOption... options);
```

Comment:
That would be my preference too but since we support a `null` content, having an overloaded `create` method makes ambiguous the following call:

```java
datastore.create(blobInfo, null);
```

Thoughts?